src/coreclr/Directory.Build.props

```diff
@@ -17,10 +17,10 @@
     <IsNativeAotProject Condition="$(MSBuildProjectDirectory.Contains('nativeaot'))">true</IsNativeAotProject>
     <BaseIntermediateOutputPath Condition="'$(IsNativeAotProject)' != 'true'">$([MSBuild]::NormalizeDirectory('$(ArtifactsObjDir)', 'coreclr', '$(MSBuildProjectName)'))</BaseIntermediateOutputPath>
     <BaseIntermediateOutputPath Condition="'$(IsNativeAotProject)' == 'true'">$([MSBuild]::NormalizeDirectory('$(ArtifactsObjDir)', 'coreclr', 'nativeaot', '$(MSBuildProjectName)'))</BaseIntermediateOutputPath>
-    <IntermediateOutputPath Condition="'$(PlatformName)' == 'AnyCPU'">$(BaseIntermediateOutputPath)$(Configuration)\</IntermediateOutputPath>
-    <IntermediateOutputPath Condition="'$(PlatformName)' != 'AnyCPU'">$(BaseIntermediateOutputPath)$(PlatformName)\$(Configuration)\</IntermediateOutputPath>
+    <IntermediateOutputPath Condition="'$(PlatformName)' == 'AnyCPU'">$(BaseIntermediateOutputPath)$(RuntimeConfiguration)\</IntermediateOutputPath>
```

Comment:
I think this is because we're now building crossgen from the packaging project, which doesn't necessarily set the same variables as the coreclr partition. I think this is more robust, although it may not be complete. Hopefully any problems should be easy to trace down, though, as the files will simply not appear in the right directories.

Reply:
It looks like the CoreCLR partition expects someone already made it so that Configuration is populated from RuntimeConfiguration. Would it be more in line with the project file expectations to set Configuration=$(RuntimeConfiguration) as an AdditionalProperty in Microsoft.NETCore.App.Crossgen2.sfxproj?

Reply:
Changed it, although I'm not sure this is robust going forward. I think it would make more sense for each partition to set the appropriate properties it needs to build, so it can be referenced transparently from anywhere, rather than have it set from the outside.

Probably a broader change though.

Reply:
> Probably a broader change though

Yep, we would also have to make sure it keeps working when opening/building the CoreCLR projects with VS (`Configuration` property has a meaning to VS).